src/app/planmylandoperation/esrimap/esrimap.component.ts

```diff
@@ -22,20 +22,20 @@ import { Subscription } from 'rxjs';
 import { HelpService } from 'src/app/shared/services/help/help.service';
 import { HelpObj } from 'src/app/shared/services/help/HelpObj.model';
 import { InitializeArcGISWorkers } from 'src/app/shared/utils/ArcGISWorkersUtil';
-import { isMapViewActive } from 'src/app/shared/ScreenUtils';
 import { GraphicsStoreComponent } from 'src/app/shared/components/graphics-store/GraphicsStore.component';
 import { defaultPointCircleSymbol } from 'src/app/shared/utils/DefaultSymbols';
 import { ElevationProfileComponent } from '../sidebar/elevation-profile/elevation-profile.component';
 import { CulvertSizeComponent } from '../sidebar/culvert-size/culvert-size.component';
 import { SoilsComponent } from '../sidebar/soils/soils.component';
 import { reorderGraphicsLayer } from 'src/app/shared/utils/LayerUtils';
+import { ListenToKeyboard } from 'src/app/shared/utils/MapViewUtils';
 
 @Component({
   selector: 'pmlo-esrimap',
   templateUrl: './esrimap.component.html',
   styleUrls: ['./esrimap.component.scss']
 })
-export class EsrimapComponent implements OnInit, AfterViewInit {
+export class EsrimapComponent implements OnInit, AfterViewInit, OnDestroy {
```

Comment:
make sure destroy lifecycle hook is fully utilized.
Unsubscribe from each and every observable at the end.